src/rsvc/srv.c

```diff
@@ -271,7 +271,7 @@ ds_rsvc_lookup(enum ds_rsvc_class_id class, d_iov_t *id,
 
 	entry = d_hash_rec_find(&rsvc_hash, id->iov_buf, id->iov_len);
 	if (entry == NULL) {
-		char	       *path;
+		char	       *path = NULL;
```

Comment:
Should we skip the 'stat()' call when path is NULL? I'm not sure if 'stat' on a NULL path will cause problem.

Reply:
Updated patch to create a dummy path in the mgmt cb for sc_locate(). The old mgmt svc implementation is mostly removed now and will eventually be completely removed.

Reply:
hmm, why not just remove mgmt_svc_locate_cb() ?

Then check 
if (rsvc_class(class)->sc_locate(id, &path))
         rsvc_class(class)->sc_locate() 
?

Reply:
Mostly just trying to avoid changing code outside of mgmt to accommodate the nodb hack I added. I want to remove that eventually and would like to keep the tech debt contained.

Reply:
ah, ok

Reply:
Why do we need to set path to NULL here? Is it just paranoid?

Reply:
Before my update to generate a dummy path in the sc_locate() cb, we would get down to the D_FREE(path) which was called on an undefined value and the server would segfault. So my first inclination was to just make sure that D_FREE() was being called on a NULL value. Then it occurred to me that it would probably be better to avoid calling stat() on an null value, hence the second commit. Technically, setting path = NULL isn't necessary anymore, but it doesn't seem to hurt anything.

In any case, I'm not proud of the hacks here, but I just don't have time to properly address the requirement of doing an asynchronous map broadcast without using this vestige of the mgmt service. If someone else has the time to look at it before 1.2 is frozen, I would be happy to get rid of the hacks, but otherwise I think we need to just use this solution because it works, even if it's not pretty.